src/instructlab/data/generate.py

```diff
@@ -214,10 +222,17 @@ def generate(
         from instructlab.model.backends.llama_cpp import Server as llama_cpp_server
 
         # TODO (cdoern): we really should not edit the cfg object
-        ctx.obj.config.serve.llama_cpp.llm_family = model_family
-        backend_instance = backends.select_backend(
-            cfg=ctx.obj.config.generate.teacher, model_path=model_path
+        serve_cfg = copy.copy(ctx.obj.config)
+        serve_cfg.generate.teacher.llama_cpp.llm_family = model_family
+        if gpus is not None:
+            serve_cfg.generate.teacher.vllm.vllm_args.extend(
+                ["--tensor-parllel-size", gpus]
+            )
+        backend_instance, backend_type = backends.select_backend(
+            cfg=serve_cfg.generate.teacher, model_path=model_path
         )
+        if backend_type is not backends.VLLM and gpus is not None:
```

Comment:
evaluate does a log debug that it's ignore this state.  We should be consistent.